src/components/PropProvider/propConnect.tsx

```diff
@@ -108,10 +115,10 @@ function propConnect(name?: ConfigGetter, options: Object = {}) {
 
         return {
           'data-cy-component': namespace,
-          'data-cy': namespace,
           ...namespacedProps,
           ...rest,
           className,
+          'data-cy': this.getDataNamespace(),
```

Comment:
Moved down to override `[data-cy]` that my exist in `...rest`. The `this.getDataNamespace()` method takes `this.props` into consideration.